cli/tests/helpers.go

```diff
@@ -0,0 +1,73 @@
+package tests
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"time"
+)
+
+// Executor allows callers to run external commands and capture their combined output.
+// Implementations should be safe for concurrent use by multiple goroutines.
+// All helpers in tests package use Executor to abstract command execution.
+//
+// The returned []byte contains stdout and stderr combined, identical to exec.Cmd.CombinedOutput().
+// The error will be non-nil if the command exits with non-zero status or fails to start.
+// Callers may use errors.As(err, *exec.ExitError) to inspect exit code if required.
+//
+// Example:
+//
+//	exec := tests.NewCmdExecutor("")
+//	out, err := exec.Execute("echo", "hello")
+//	fmt.Printf("%s", out) // prints "hello\n"
+type Executor interface {
```

Comment:
can we use the exec package directly here without adding the Executor abstraction? Will we support different types of executors? I feel we would mostly need just a os command executor..we can expose a helper function called `ExecuteCommand` or something..this feels simpler to me

Reply:
Testing becomes much easier when you have an interface for Executor. You can use a test implementation and inject this mocked dependency when need arises in future.